Quick_Deploy/vLLM/client.py

```diff
@@ -0,0 +1,193 @@
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import asyncio
+import json
+import queue
+import sys
+import uuid
+
+import numpy as np
+import tritonclient.grpc.aio as grpcclient
+from tritonclient.utils import *
+
+
+class UserData:
+    def __init__(self):
+        self._completed_requests = queue.Queue()
+
+
+def get_request_data(prompt, temperature, top_p):
+    request_json = {
+        "prompt": prompt,
+        "temperature": temperature,
+        "top_p": top_p,
+        "echo": True,
+    }
+    return np.array([json.dumps(request_json).encode("utf-8")], dtype=np.object_)
+
+
+async def async_stream_yield(
+    prompt_tokens, temperature, top_p, sequence_id, model_name
+):
+    count = 1
+    for token in prompt_tokens:
+        # Create the tensor for INPUT
+        request_data = get_request_data(token, temperature, top_p)
+        inputs = []
+        inputs.append(
+            grpcclient.InferInput(
+                "serialized_request_json", request_data.shape, "BYTES"
+            )
+        )
+        # Initialize the data
+        inputs[0].set_data_from_numpy(request_data)
+        outputs = []
+        outputs.append(grpcclient.InferRequestedOutput("serialized_response_json"))
+        # Issue the asynchronous sequence inference.
+        yield {
+            "model_name": model_name,
+            "inputs": inputs,
+            "outputs": outputs,
+            "request_id": "{}_{}".format(sequence_id, count),
+            "sequence_id": sequence_id,
+            "sequence_start": (count == 1),
+            "sequence_end": (count == len(prompt_tokens)),
+        }
+        count = count + 1
+
+
+async def main(FLAGS):
+    model_name = "vllm"
+    temperature = 0.8
+    top_p = 0.95
+    prompt_list = [
+        ["Hello, my name is"],
+        ["The most dangerous animal is"],
+        ["The capital of France is"],
+        ["The future of AI is"],
+    ]
+
+    async with grpcclient.InferenceServerClient(
+        url=FLAGS.url, verbose=FLAGS.verbose
+    ) as triton_client:
+        # Request iterator that yields the next request
+        async def async_request_iterator():
+            prompt_id = FLAGS.offset
+            for prompt in prompt_list:
+                prompt_id = prompt_id + 1
+                async for request in async_stream_yield(
```

Comment:
once each prompt is a string we can simplify I think to just a synchronous "create_request" method here to create the request from the prompt